File: lib/core/Dimensions.ts
export default class Dimensions {
  From: number;
  To: number;

  public static MinMax(): Dimensions {
    return new Dimensions(Number.MAX_VALUE, Number.MIN_VALUE);
  }

  constructor(from: number, to: number) {
    this.From = from;
    this.To = to;
  }

  public static add(x: Dimensions, y: Dimensions): Dimensions {
    return new Dimensions(Math.min(x.From, y.From), Math.max(x.To, y.To));
  }
}

File: lib/core-cs/Dimensions.cs
/*
 * Copyright (c) Roman Polunin 2016. 
 * MIT license, see https://opensource.org/licenses/MIT. 
*/
using System;

namespace OrgChart.Layout
{
    /// <summary>
    /// Edges of a bunch of siblings on vertical or horizontal axis.
    /// </summary>
    public struct Dimensions
    {
        /// <summary>
        /// Min value.
        /// </summary>
        public readonly double From;
        /// <summary>
        /// Max value.
        /// </summary>
        public readonly double To;

        /// <summary>
        /// Ctr.
        /// </summary>
        public static Dimensions MinMax()
        {
            return new Dimensions(double.MaxValue, double.MinValue);
        }
        
        /// <summary>
        /// Ctr.
        /// </summary>
        public Dimensions(double from, double to)
        {
            From = from;
            To = to;
        }

        /// <summary>
        /// Computes combined dimension.
        /// </summary>
        public static Dimensions operator +(Dimensions x, Dimensions y)
        {
            return new Dimensions(Math.Min(x.From, y.From), Math.Max(x.To, y.To));
        }
    }
}